ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryExecutor.java

```diff
@@ -152,7 +143,8 @@ public TransientQueryMetadata buildTransientQuery(
       final LogicalSchema schema,
       final OptionalInt limit,
       final Optional<WindowInfo> windowInfo,
-      final boolean excludeTombstones
+      final boolean excludeTombstones,
+      final QueryMetadata.Listener listener
```

Comment:
Personally I found passing in this listener, which is not supposed to be changed in production, is a bit odd -- maybe for testing purposes we can expose a setter to pass in a mock after the queryMetadata is created? Anyways, if this is the pattern we've followed for testing for a while feel free to ignore it.

Reply:
Yeah I agree it's a bit strange. I prefer not to add setters and have whatever's needed passed in when the query is created. I guess this class could really be put into QueryRegistryImpl - it's really only meant to be used from there. I opted not to to keep the registry focused. At minimum we can make this class package-private.